docs/usage/customsessions.md

```diff
@@ -0,0 +1,145 @@
+# Create a `CustomSessionStorage` Solution
+
+This library comes with two session management options: `MemorySessionStorage` and `CustomSessionStorage`.
+
+`MemorySessionStorage` exists as an option to help you get started developing your apps as quickly as possible, and is the default storage option on `Shopify.Context`. It's perfect for working in your development and testing environments. However, this storage solution is not meant to be used in production due to its limitations.
+
+When you're ready to deploy your app and run it in production, you'll need to set up a `CustomSessionStorage`, which you can then use in initializing your `Shopify.Context`. The `CustomSessionStorage` class expects to be initialized with three callbacks that link to your chosen storage solution and map to the `storeSession`, `loadSession`, and `deleteSession` methods on the class.
+
+## Callback Methods
+
+- All of the callbacks used to create a new instance of `CustomSessionStorage` should be `async` functions and return a `Promise` that resolves to a specified type, as outlined below.
+
+| Method           | Arg Type  | Return Type                                                 | Notes                                                                                                                                                                                                                |
+| ---------------- | --------- | ----------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```

Comment:
Perhaps it's because I'm reviewing this on GitHub where the lines wrap — I could see going either way on this, tbh — BUT personally I find it more readable to shorten the `thead` elements to their most minimal format:

```suggestion
| Method | Arg type | Return type | Notes |
| --- | --- | --- | --- |
```

Also it means that you don't have to constantly adjust the widths of the header and divider cells when the content changes later.

Finally! Table column headers are also sentence case.